ui/src/components/PortfolioForm.tsx

```diff
@@ -141,57 +134,57 @@ export default function PortfolioForm() {
           </h3>
 
           <div className="setting-row">
-            <label>Stake when tokens arrive:</label>
-            <input
-              type="checkbox"
-              checked={settings.stake || false}
-              onChange={() => toggleSetting('stake')}
-            />
+            <label>Polling Frequency:</label>
           </div>
           <div className="button-group">
             <button
               type="button"
-              disabled={!settings.stake}
-              className={settings.stakeFrequency === 'daily' ? 'active' : ''}
-              onClick={() => setFrequency('stake', 'daily')}
+              className={config.freq === 'daily' ? 'active' : ''}
+              onClick={() =>
+                updateChainConfig(selectedChain, {
+                  freq: config.freq === 'daily' ? '' : 'daily',
+                })
+              }
             >
               Daily
             </button>
             <button
               type="button"
-              disabled={!settings.stake}
-              className={settings.stakeFrequency === 'weekly' ? 'active' : ''}
-              onClick={() => setFrequency('stake', 'weekly')}
+              className={config.freq === 'weekly' ? 'active' : ''}
+              onClick={() =>
+                updateChainConfig(selectedChain, {
+                  freq: config.freq === 'weekly' ? '' : 'weekly',
+                })
+              }
             >
               Weekly
             </button>
           </div>
 
           <div className="setting-row">
-            <label>Restake when rewards accrue:</label>
+            <label>Stake when tokens arrive:</label>
             <input
               type="checkbox"
-              checked={settings.restake || false}
-              onChange={() => toggleSetting('restake')}
+              checked={!!config.onReceipt.length}
+              onChange={() =>
+                updateChainConfig(selectedChain, {
+                  onReceipt: config.onReceipt.length ? [] : ['stake'],
+                })
+              }
             />
           </div>
-          <div className="button-group">
-            <button
-              type="button"
-              disabled={!settings.restake}
-              className={settings.restakeFrequency === 'daily' ? 'active' : ''}
-              onClick={() => setFrequency('restake', 'daily')}
-            >
-              Daily
-            </button>
-            <button
-              type="button"
-              disabled={!settings.restake}
-              className={settings.restakeFrequency === 'weekly' ? 'active' : ''}
-              onClick={() => setFrequency('restake', 'weekly')}
-            >
-              Weekly
-            </button>
+
+          <div className="setting-row">
+            <label>Restake when rewards accrue:</label>
```

Comment:
To give the impression that we may add other features, I **suggest**:

When rewards accrue:
 - [ ] restake